wemake_python_styleguide/visitors/ast/redundancy.py

```diff
@@ -53,3 +54,51 @@ def _check_for_redundant_enumerate(self, node: Union[AnyFor, ast.comprehension])
 
         if isinstance(index_receiver, ast.Name) and index_receiver.id == '_':
             self.add_violation(RedundantEnumerateViolation(index_receiver))
+
+
+@final
+class RedundantTernaryVisitor(BaseNodeVisitor):
+    """Finds useless ternary operators."""
+
+    def visit_IfExp(self, node: ast.IfExp) -> None:
+        """Finds useless ternary operators."""
+        body = ast.unparse(node.body)
+        orelse = ast.unparse(node.orelse)
+
+        # Check for specific patterns.
+        self.body_orelse_check(node, body, orelse)
+        self.check_computed_equal(node)
+
+        self.generic_visit(node)  # Visit all other node types.
+
+    def body_orelse_check(self, node, body, orelse) -> None:
+        """Check if body and orelse are the same."""
+        if body == orelse:
+            self.add_violation(RedundantTernaryViolation(node))
+
+    def check_computed_equal(self, node) -> None:
+        """Used to check if the computed branches are equal."""
+        if not isinstance(node.test, ast.Compare):
+            return
+
+        allowed_ops = (ast.NotEq, ast.Eq, ast.IsNot)
+        is_node_test = isinstance(node.test, ast.Compare)
+        correct_op = any(
+            isinstance(node.test.ops[0], op) for op in allowed_ops
+        )
+
+        if is_node_test and correct_op and len(node.test.ops) == 1:
+            left = ast.unparse(node.test.left)
+            right = ast.unparse(node.test.comparators[0])
+
+            self.compare_operands(node, left, right)
+
+    def compare_operands(self, node, left, right) -> None:
```

Comment:
please, type `node`, `left`, and `right`

Reply:
By this did you mean the type hints?

Reply:
yes